### ISSUE TITLE
[FR] Automatically capture reports from ./gradlew --profile on CI

### ISSUE THREAD

**Issue by iamdanfox** [palantir/gradle-circle-style#28]
Running `./gradlew --profile <task>` locally is a super helpful way to diagnose slowness, but it seems to always output the nice HTML report to the [`build/reports/profile` directory](https://github.com/gradle/gradle/blob/v4.8.1/subprojects/core/src/main/java/org/gradle/profile/ReportGeneratingProfileListener.java#L48).

I'd like to add `--profile` to all my tasks on CI because profiling is pretty cheap and it's interesting to be able to click through and see whether there's low-hanging fruit.  Currently, I'd need to add a bunch of nasty lines of bash to copy these reports into the `$CIRCLE_ARTIFACTS` directory.

As an alternative, what do you think about making this gradle plugin detect the `--profile` flag and add it's own version of the `ReportGeneratingProfileListener` which writes to the `$CIRCLE_ARTIFACTS` directory?

**Comment by alicederyn** [comment]
Sounds cheap and smart. Definitely within the scope of this plugin. Only problem is develop is stuck on a blocked feature. Base your PR off the 1.x branch and I'll fix that independently.

**Comment by iamdanfox** [comment]
Cool, can't promise a PR soon but glad you think this repo could be a good home for this feature :)

**Comment by alicederyn** [comment]
I would expect you to be able to configure the target directory with a single line of Gradle by the way, definitely check to see if that's possible before sinking time into the listener route (assuming you haven't already search and hit a blank).

**Comment by iamdanfox** [comment]
Yeah I was really hoping there would be a one-liner to do that too, but my googling skills are apparently not good enough (and as far as I can tell, the `build/reports/profile` directory [isn't configurable](https://github.com/gradle/gradle/blob/v4.8.1/subprojects/core/src/main/java/org/gradle/profile/ReportGeneratingProfileListener.java#L48))

**Comment by alicederyn** [comment]
You just *catapulted* past the bar for becoming an admin on this repository, holy cow.
